lib/sentry/client_report.ex

```diff
@@ -0,0 +1,128 @@
+defmodule Sentry.ClientReport do
+  @moduledoc """
+  **TODO** add proper doc for module
+
+  See <https://develop.sentry.dev/sdk/client-reports/>.
+  """
+
+  @moduledoc since: "10.0.0"
+
+  @typedoc since: "10.0.0"
+
+  use GenServer
+  alias Sentry.Client
+
+  @typedoc """
+  The possible reasons of the discarded event.
+  """
+  @type reasons() ::
+          :ratelimit_backoff
+          | :queue_overflow
+          | :cache_overflow
+          | :network_error
+          | :sample_rate
+          | :before_send
+          | :event_processor
+          | :insufficient_data
+          | :backpressure
+
+  @typedoc """
+  The struct for a **client report** interface.
+  """
+  @type t() :: %__MODULE__{
+          timestamp: String.t() | number(),
+          discarded_events:
+            list(%{reason: reasons(), category: String.t(), quantity: pos_integer()})
```

Comment:
Nit, but `list/1` is somewhat less common than just:

```suggestion
            [%{reason: reasons(), category: String.t(), quantity: pos_integer()}, ...]       
```

Also added `, ...` because a client report with no discarded events makes no sense, does it? (Honest question, I really don't know)